### PR TITLE
feat: Local build

### PR DESCRIPTION
**Pull request by Alystrasz**
Instead of relying on `github-pages` to render the website hosted in this repository, this adds instructions to build and display the website locally, which is more convenient during development.
This should have no impact on the `github-pages` deployment.

Sans titre

This adds:
* A `Gemfile` configuration file used for local deployment;
* Documentation on how to use it.

### CODE REVIEW

**Comment by basvdheuvel** [comment]
Hi, great work, thanks!
One question/remark: should we not include `Gemfile` and `Gemfile.lock` in the repo?
This way everyone (including the github.io build actions!) should be running the same versions of packages, and updates are explicitly commited to the server.
What do you think?

**Comment by Alystrasz** [comment]
I agree it would be the cleanest way to do it; actually I thought about doing that at first, but then thought I couldn't run the build action locally and thus had no idea of the effect of such a modification on the github action (and I definitely didn't want to break the `https://www.discotec.org/` deployment with my first PR :smile:)

**Comment by basvdheuvel** [comment]
My personal github.io website has been running like that for years without issue (even better, GH alerts me of security vulnerabilities in outdated packages).

I do remember there being some issues with `github-pages` plugins when I was building the site locally myself (things like automatically giving the page a title).
I lost the Gemfile I wrote to fix that, but later today I'll have a look if I can find these things. I'll then add the commit to this PR, so you can test my Gemfile and we can hopefully merge.

**Comment by basvdheuvel** [comment]
Okay, please have a look.

File: _config.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+theme: jekyll-theme-primer
```
Comment (Alystrasz) [inline _config.yml:1]: So is `jekyll-theme-primer` the style used by this website?
Another theme seems to be defined in the Gemfile:
```shell
# This is the default theme for new Jekyll sites. You may change this to anything you like.
gem "minima", "~> 2.0"
```

**Review by Alystrasz — COMMENTED** [review]
Unsure which causes this, but the title of the main page does not appear anymore with latest modifications:

#### `discotec.org` deployed version
image

#### Local version
image

**Comment by basvdheuvel** [comment]
Okay there was some confusion on my side, but I think all is well.
Github pages uses `jekyll-theme-primer` by default, so I set that in `_config.yml`.
Headings are fixed by disabling `titles_from_headings` plugin.

**Comment by Alystrasz** [comment]
I don't see anymore issues; I say we're good to go!

**Pull request merged**